core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -554,8 +558,10 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                     "REFRESH MATERIALIZED VIEW",
                     targetTable,
                     Optional.empty(),
-                    Optional.of(insertColumns.stream()
-                            .map(insertColumn -> new Column(insertColumn, tableMetadata.getColumn(insertColumn).getType().toString()))
+                    Optional.of(Streams.zip(
+                            insertColumns.stream(),
+                            queryScope.getRelationType().getVisibleFields().stream(),
+                            this::createOutputColumn)
```

Comment:
The new method uses `field.getType()` for the `Column` type, instead of `tableMetadata.getColumn(insertColumn).getType()`. The types of are not necessarily the same (see `typesMatchForInsert `)

Reply:
Yeah !! But we need also `Field` for fetching the lineage. So instead of fetching from the `tableMetadata` we thought of fetching it from `Field`

Reply:
All right, I understand that you need to fetch the source columns from field. But besides that, there's another unrelated detail that changed. It is the Type used for the `Column` construction. It should be taken from the column metadata, not from the field.

Reply:
Oh okay. Have addressed it. Thanks for the pointer.